apollo-api/src/commonMain/kotlin/com/apollographql/apollo/api/SealedEnumType.kt

```diff
@@ -0,0 +1,11 @@
+package com.apollographql.apollo.api
+
+/**
+ * Base class for an enum type when generating sealed Kotlin class
+ */
+abstract class SealedEnumType {
```

Comment:
I kind of wanted to follow `ScalarType` naming, but of course can change it to whatever is more appropriate.

Also, @martinbonnin suggested using an interface (https://github.com/apollographql/apollo-android/issues/2775#issuecomment-734399072), but I wasn't sure it's okay to have a default method with `toString` implementation

Reply:
> I kind of wanted to follow ScalarType naming, but of course can change it to whatever is more appropriate.

Naming is hard... I'm personnaly not a fan of `ScalarType` as it acts more like a type-safe `Binding` or `Mapping` between a GraphQL type and a Kotlin one. So I don't mind not following the logic here. 

What about the below:
```
interface EnumValue {
  val rawValue: String
}
```

? This will be closest to the spec: https://spec.graphql.org/draft/#EnumValue

> And if the idea was to add a case here then of course I can also do that, it's just overriding toString feels like a good solution

It was the original intent. I feel it's dangerous to rely on `toString()` because it typically exposes to these kind of bugs that we're seeing here. The compiler cannot catch if a new type is added a someone forgets to override `toString()`. 

Ultimately, I was thinking of moving this:
```
fun writeToJson(value: Any?, jsonWriter: JsonWriter) {
```
to something more like:
```
fun writeToJson(value: InputValue, jsonWriter: JsonWriter) {
```

where `InputValue` would be an instance of an InputType as defined by https://spec.graphql.org/draft/#IsInputType(), i.e. scalar, enum or InputObject

Reply:
I changed the abstract class to `interface EnumValue`, following the GraphQL spec definitely makes sense! Also with that change I now could implement that interface for all enum classes, not only the ones generated as sealed classes.

`writeToJson(value: InputValue` also makes perfect sense. I originally thought it's more of a general-purpose utility, that's why I had doubts :)